File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjJavaFX/Services/src/main/java/app/services/IServices.java
package app.services;

import app.model.Employee;
import app.model.Game;
import app.model.TicketPurchase;

import java.util.List;

public interface IServices {
    void logIn(String username, String password, IObserver client) throws MyException;
    void purchase(long gameId, int noOfTickets, String clientName) throws MyException;
    void logOut(Employee employee, IObserver client) throws MyException;
    List<Game> findAllGames() throws MyException;
}

File: sem4/MPP/mpp-proiect-repository-mariaruncan/ProjCSharp/Services/IServices.cs
using System.Collections.Generic;
using Model;

namespace Services
{
    public interface IServices
    {
        void LogIn(string username, string pass, IObserver client);
        void Purchase(long gameId, int noOfTickets, string clientName);
        void LogOut(Employee employee, IObserver client);
        List<Game> FindAllGames();
    }
}
